### PR TITLE
Add rudimentary support for connecting through a ssh tunnel

### PR DESCRIPTION
**Pull request by cosimo**
For Couchbase clusters in certain environments, we don't allow direct access to the rest api admin port (8091, 18091).
Here I tried to implement a ssh-tunnel connection option. When specified, this would automatically open a tunnel to a local port via ssh, and then proceed as usual.

Doing so would allow to setup clusters without having direct access to port 8091.

### CODE REVIEW

File: couchbase_cluster_admin/cluster.py [hunk 1]
```diff
@@ -41,6 +44,25 @@ def __init__(
         self.username = username
         self.password = password
 
+        if connect_through_ssh:
+            if not ssh_username:
+                raise ValueError("You need to specify a `ssh_username`")
+
+            self.ssh_tunnel = SshTunnel(
```
Comment (sveniu) [inline couchbase_cluster_admin/cluster.py:50]: This doesn't specify local_bind_address, which seems to default to `('0.0.0.0', RANDOM_PORT)`, possibly exposing the endpoint to other clients in the network. It looks like you can specify `('127.0.0.1',)` to get the automatic port.

File: couchbase_cluster_admin/cluster.py [hunk 2]
```diff
@@ -41,6 +44,25 @@ def __init__(
...
+                ssh_username=ssh_username,
+                remote_host=api_host,
+                remote_port=api_port,
+            )
+            self.ssh_tunnel.start()
+            self.api_host = "127.0.0.1"
```
Comment (sveniu) [inline couchbase_cluster_admin/cluster.py]: I guess this is always `127.0.0.1`? The api seems to expose both [local_host and local_port](https://github.com/pahaz/sshtunnel/blob/aa7342e55e49529fea020a09ba8f1eed33eb896f/sshtunnel.py#L411-L417) to obain the local endpoint. Dunno how relevant it is with non-127.0.0.1 local hosts..

File: couchbase_cluster_admin/cluster.py (hunk 1 above)
Comment (cosimo) [reply couchbase_cluster_admin/cluster.py:50]: Modified now to use `('127.0.0.1',)` as you suggested.

**Comment by cosimo** [comment]
Rebasing an active PR is against my core values, but I did it anyway... :crossed_fingers:

**Pull request merged**